### PR TITLE
Changelog for Rust 1.70 :hammer:

### PR DESCRIPTION
**Pull request by xFrednet**
Roses are red,
violets are blue,
damn I have an exam to cram,
and this rhyme is a scam

---

This poem is... certainly something... Anyways, hope whoever is reading this, has a lovely day full of sunshine without the need to study :D

---

changelog: none

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,7 +6,132 @@ document.
...
+* [`uninit_vec`], [`uninit_assumed_init`]: Now uses a better heuristic
+  [#10520](https://github.com/rust-lang/rust-clippy/pull/10520)
+* [`ifs_same_cond`]: Now also detects immutable method calls.
+  [#10350](https://github.com/rust-lang/rust-clippy/pull/10350)
+* [`arithmetic_side_effects`]: No longer lints on right or left shifts with constant integers, as the
+  compiler warns about them
+  [#10309](https://github.com/rust-lang/rust-clippy/pull/10309)
+* [`items_after_statements`]: `#[allow(items_after_statements)]` now works on items
+  [#10542](https://github.com/rust-lang/rust-clippy/pull/10542)
+* [`significant_drop_tightening`]: was optimized
```
Comment (giraffate) [inline CHANGELOG.md:73]:
```suggestion
* [`significant_drop_tightening`]: Was optimized
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,132 @@ document.
...
+  [#10589](https://github.com/rust-lang/rust-clippy/pull/10589)
+* [`manual_clamp`]: No longer lints in constant context
+  [#10479](https://github.com/rust-lang/rust-clippy/pull/10479)
+* [`almost_swapped`]: Now ignores external macros
+  [#10502](https://github.com/rust-lang/rust-clippy/pull/10502)
+* [`nonminimal_bool`]: Now ignores macros
+  [#10527](https://github.com/rust-lang/rust-clippy/pull/10527)
+* [`needless_return`]: No longer lints match statements with incompatible branches
+  [#10593](https://github.com/rust-lang/rust-clippy/pull/10593)
+* [`use_self`]: do not suggest using `Self` in const generic parameters
```
Comment (giraffate) [inline CHANGELOG.md:94]:
```suggestion
* [`use_self`]: Do not suggest using `Self` in const generic parameters
```

**Comment by llogiq** [comment]
Don't you worry 'bout your rhyme,
a good one will come in due time,
and in the meantime you'll arrange
to chronicle this version's change.

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Pull request merged**